.github/workflows/ci.yml

```
@@ -14,116 +14,124 @@ concurrency:
+          - runs-on: macos-15
+            compiler: gcc-14
+          - runs-on: macos-15
+            compiler: gcc-15
+    uses: munich-quantum-toolkit/workflows/.github/workflows/reusable-cpp-tests-macos.yml@matrix-strategy
     with:
-      # Runs to enable
-      enable-ubuntu2404-gcc-release: true
-      enable-ubuntu2404-arm-gcc-release: true
-      enable-macos13-clang-release: true
-      enable-macos14-clang-release: true
-      enable-windows2022-msvc-release: true
-      enable-windows11-arm-msvc-release: true
-      # Runs to disable
-      enable-ubuntu2404-gcc-debug: false
-      enable-ubuntu2404-clang-release: false
-      enable-ubuntu2404-clang-debug: false
-      enable-ubuntu2204-gcc-release: false
-      enable-ubuntu2204-gcc-debug: false
-      enable-ubuntu2204-clang-release: false
-      enable-ubuntu2204-clang-debug: false
-      enable-ubuntu2404-arm-gcc-debug: false
-      enable-ubuntu2404-arm-clang-release: false
-      enable-ubuntu2404-arm-clang-debug: false
-      enable-ubuntu2204-arm-gcc-release: false
-      enable-ubuntu2204-arm-gcc-debug: false
-      enable-ubuntu2204-arm-clang-release: false
-      enable-ubuntu2204-arm-clang-debug: false
-      enable-macos13-clang-debug: false
-      enable-macos13-gcc-release: false
-      enable-macos13-gcc-debug: false
-      enable-macos14-clang-debug: false
-      enable-macos14-gcc-release: false
-      enable-macos14-gcc-debug: false
-      enable-macos15-clang-release: false
-      enable-macos15-clang-debug: false
-      enable-macos15-gcc-release: false
-      enable-macos15-gcc-debug: false
-      enable-windows2022-msvc-debug: false
-      enable-windows2022-clang-release: false
-      enable-windows2022-clang-debug: false
-      enable-windows2025-msvc-release: false
-      enable-windows2025-msvc-debug: false
-      enable-windows2025-clang-release: false
-      enable-windows2025-clang-debug: false
-      enable-windows11-arm-msvc-debug: false
-      enable-windows11-arm-clang-release: false
-      enable-windows11-arm-clang-debug: false
-      # General settings
-      cmake-args-macos: -DMQT_CORE_WITH_GMP=ON
+      runs-on: ${{ matrix.runs-on }}
+      compiler: ${{ matrix.compiler }}
+      config: ${{ matrix.config }}
+      cmake-args: -DMQT_CORE_WITH_GMP=ON
 
-  # run extensive C++ tests whenever this is on a PR and the PR has the `extensive-cpp-ci` label
-  cpp-tests-extensive:
-    name: 🇨‌ Test (Extensive)
+  # run extensive C++ tests on PRs labeled with the `extensive-cpp-ci` label
+  cpp-tests-extensive-windows:
+    name: 🇨‌ Test (Extensive) 🏁
     needs: change-detection
     if: fromJSON(needs.change-detection.outputs.run-cpp-tests) && github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'extensive-cpp-ci')
-    uses: munich-quantum-toolkit/workflows/.github/workflows/reusable-cpp-ci.yml@v1.12
+    strategy:
+      fail-fast: false
+      matrix:
+        runs-on: [windows-2022, windows-11-arm]
```

This is missing windows 2025
```suggestion
        runs-on: [windows-2022, windows-2025, windows-11-arm]
```